qiita_db/metadata_template.py

```diff
@@ -135,14 +135,14 @@ def _as_python_types(metadata_map, headers):
     return values
 
 
-def _prefix_sample_names_with_id(md_template, study):
+def _prefix_sample_names_with_id(md_template, study_id):
```

Comment:
What happens if they pass the wrong study for the existing prep template? This way you have automatic data integrity since the study id comes from the object itself.